testkit/mockstore.go

```diff
@@ -17,20 +17,42 @@
 package testkit
 
 import (
+	"flag"
 	"testing"
 	"time"
 
 	"github.com/pingcap/tidb/domain"
 	"github.com/pingcap/tidb/kv"
 	"github.com/pingcap/tidb/session"
+	"github.com/pingcap/tidb/store/driver"
 	"github.com/pingcap/tidb/store/mockstore"
 	"github.com/stretchr/testify/require"
 	"github.com/tikv/client-go/v2/oracle"
 	"github.com/tikv/client-go/v2/tikv"
 )
 
+// WithTiKV flag makes the test case run with real TiKV
+var WithTiKV = flag.String("with-tikv", "", "whether tests run with real TiKV")
```

Comment:
In fact, @tisonkun has moved the real-tikv test case on [here](https://github.com/pingcap/tidb/blob/master/tests/realtikvtest/testkit.go#L43). If you would like to run the real-tikv test case. all are under the same folder.

Reply:
The background is, I need to test the paging protocol with the real tikv.
So I need rewrite all the test case or move them all?
A more reasonable solution is just add a flag ... if you don't use it, it doesn't change anything.

Reply:
@tiancaiamao would you like to change CI logic? Or adding these code for conveniently running tests with real tikv locally?

Reply:
Also, `with-tikv` is a string flag while its help info said `"whether tests run with real TiKV"`, which is confusing...

Perhaps

```go
// This flag is only used for debugging locally with real tikv cluster.
var tikvAddr = flag.String("tikv-addr", "", "address of tikv cluster, if set, running test with real tikv cluster")
```

Reply:
> Also, `with-tikv` is a string flag while its help info said `"whether tests run with real TiKV"`, which is confusing...
> 
> Perhaps
> 
> ```go
> // This flag is only used for debugging locally with real tikv cluster.
> var tikvAddr = flag.String("tikv-addr", "", "address of tikv cluster, if set, running test with real tikv cluster")
> ```

Updated.

Reply:
> @tiancaiamao would you like to change CI logic? Or adding these code for conveniently running tests with real tikv locally?

If you want to see the full changes, it's in my branch https://github.com/tiancaiamao/tidb/tree/test-with-tikv
But not all the test cases can run under TiKV, I check and document the result here:

https://pingcap.feishu.cn/docx/doxcnEzyRXsTdCJ2EE6EZwIP2oe

I'm not adding a new CI pipeline, I just make the code base more friendly for those who want to run some test cases with a real TiKV.